webapp/channels/src/components/new_channel_modal/new_channel_modal.tsx

```diff
@@ -217,42 +213,8 @@ const NewChannelModal = () => {
         }
     };
 
-    const handleOnDisplayNameChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-        e.preventDefault();
-        const {target: {value: displayName}} = e;
-
-        const displayNameErrors = validateDisplayName(displayName);
-
-        setDisplayNameError(displayNameErrors.length ? displayNameErrors[displayNameErrors.length - 1] : '');
-        setDisplayName(displayName);
-        setServerError('');
-
-        if (!urlModified) {
-            setURL(cleanUpUrlable(displayName));
-            setURLError('');
-        }
-    };
-
-    const handleOnDisplayNameBlur = () => {
-        if (displayName && !url) {
-            setURL(crypto.randomBytes(16).toString('hex'));
-        }
-        if (!displayNameModified) {
-            setDisplayNameModified(true);
-        }
-    };
-
-    const handleOnURLChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-        e.preventDefault();
-        const {target: {value: url}} = e;
-
-        const cleanURL = url.toLowerCase().replace(/\s/g, '-');
-        const urlErrors = validateChannelUrl(cleanURL, intl) as string[];
-
-        setURLError(urlErrors.length ? urlErrors[urlErrors.length - 1] : '');
-        setURL(cleanURL);
-        setURLModified(true);
-        setServerError('');
+    const handleOnURLChange = (updatedURL: string) => {
```

Comment:
0/5 passing a new function on every render to the child component.

Reply:
Done.